File: 0111.爬楼梯/solution.py
class Solution:
    """
    @param n: An integer
    @return: An integer
    """
    def climbStairs(self, n):
        # write your code here
        if n <= 2:
            return n
        result=[1,2]
        for i in range(n-2):
            result.append(result[-2]+result[-1])
        return result[-1]

File: 0111.爬楼梯/solution.java
public class Solution {
    /**
     * @param n: An integer
     * @return: An integer
     */
    public int climbStairs(int n) {
        // write your code here
        if (n <=1){
            return 1;
        }
        int last = 1, lastlast = 1;
        int now = 0;
        for (int i = 2;i <=n; i++){
            now = last +lastlast;
            lastlast = last;
            last = now;
        }
        return now;
    }
}
